### PR TITLE
Fixed #15143 -- Documented how to set the language for the test client.

### PR DESCRIPTION
**Pull request by mscott250**
Added a new section to the testing tools topic documentation, regarding how to set the language for a test client request in a test case.

### CODE REVIEW

File: docs/topics/testing/tools.txt [hunk 1]
```diff
@@ -570,6 +570,67 @@ can access these properties as part of a test condition.
...
+
+        # explicitly set the language for the active session
+        user_language = 'fr'
+        translation.activate(user_language)
+
+        response = self.client.get('/polls/')
+        actual_content = response.content
+
+        # deactivate the set language to return to the default behaviour
+        translation.deactivate()
```
Comment (charettes) [inline docs/topics/testing/tools.txt]: Great docs! May I suggest using the `translation.override` context manager instead?

```python
with translation.override(user_language):
    response = self.client.get('/polls/')
```

It should prevent the globally defined language from leaking if the test was to fail before `deactivate` is called.

File: docs/topics/testing/tools.txt [hunk 2]
```diff
@@ -570,6 +570,67 @@ can access these properties as part of a test condition.
...
+        self.assertEqual(actual_content, expected_content)
+
+    def test_language_using_header(self):
+        expected_content = b"Bienvenue sur mon site."
+
+        user_language = 'fr'
+        # set the desired language via the Accept-Language HTTP header
+        response = self.client.get('/polls/',
+                                   None,
+                                   HTTP_ACCEPT_LANGUAGE=user_language)
```
Comment (charettes) [inline docs/topics/testing/tools.txt]: We usually favor the following indentation

```python
response = self.client.get(
    '/polls/',
    HTTP_ACCEPT_LANGUAGE=user_language,
)
```
Note that you don't need to pass `None` as it's already the default value for the `data` kwarg.

**Comment by charettes** [comment]
It looks like the actual ticket is [#15143](https://code.djangoproject.com/ticket/15143) and not [#14143](https://code.djangoproject.com/ticket/14143). Please update you commit message to match this PR's title if you are able. Else the commiter will take care of it.

**Comment by charettes** [comment]
Please uncheck _patch needs improvement_ on the [ticket](https://code.djangoproject.com/ticket/15143) when you manage to find time to address my comments. Thanks again for your contribution.

**Comment by timgraham** [comment]
Updated in https://github.com/django/django/pull/7238.

**Pull request closed without merging**